examples/decomposition/plot_sparsenet.py

```diff
@@ -0,0 +1,135 @@
+
+"""
+=======================================
+Dictionary learning using Sparse coding
+=======================================
+
+This example compares the set of filters obtained by sparse coding 
+algorithm described by Olshausen and Field in Nature, vol. 381,
+pp. 607-609 with that obtained using :ref:`DictionaryLearning`. The
```

Comment:
I based my writing on the `plot_image_denoising.py` script which is quite similar.